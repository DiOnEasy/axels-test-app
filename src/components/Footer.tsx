import React, { FC } from 'react';

import { StyledFooter } from 'styled/StyledFooter';

export const Footer: FC = () => {
    const currentYear = new Date().getFullYear();
    return <StyledFooter>&copy; 2018-{currentYear}</StyledFooter>;
};
