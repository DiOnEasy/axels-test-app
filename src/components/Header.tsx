import React, { FC } from 'react';

import { StyledHeader } from 'styled/StyledHeader';

export const Header: FC = () => <StyledHeader>Test app</StyledHeader>;
