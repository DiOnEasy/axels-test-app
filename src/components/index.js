export { Footer } from 'components/footer/Footer';
export { Header } from 'components/header/Header';
export { GalleryImage } from 'components/galleryPage/galleryImage/GalleryImage';
export { ImageModal } from 'components/galleryPage/imageModal/ImageModal';
export { ModalButton } from 'components/galleryPage/modalButton/ModalButton';
export { ModalInput } from 'components/galleryPage/modalInput/ModalInput';
export { CommentForm } from 'components/galleryPage/commentForm/CommentForm';
