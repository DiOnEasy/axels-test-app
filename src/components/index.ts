export { Footer } from 'components/Footer';
export { Header } from 'components/Header';
export { GalleryImage } from 'components/galleryPage/GalleryImage';
export { ImageModal } from 'components/galleryPage/ImageModal';
export { ModalButton } from 'components/galleryPage/ModalButton';
export { ModalInput } from 'components/galleryPage/ModalInput';
export { CommentForm } from 'components/galleryPage/CommentForm';
