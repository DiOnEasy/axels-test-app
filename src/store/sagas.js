import { all } from '@redux-saga/core/effects';
import { watcherGetImages } from './ducks/images';

export default function* rootSaga() {
    yield all([watcherGetImages()]);
}
