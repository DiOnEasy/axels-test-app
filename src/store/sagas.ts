import { all } from '@redux-saga/core/effects';

import {
    watcherFetchImageDetails,
    watcherFetchImages,
    watcherSendComment
} from './ducks/images';

export default function* rootSaga() {
    yield all([
        watcherFetchImages(),
        watcherFetchImageDetails(),
        watcherSendComment()
    ]);
}
