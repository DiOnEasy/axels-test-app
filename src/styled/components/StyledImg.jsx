import styled from "styled-components";

export const StyledImg = styled.img.attrs((props) =>({
  src:  props.src || null,
}))`
  width: 100%;
`;
