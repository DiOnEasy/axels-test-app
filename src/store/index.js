import { configureStore } from '@reduxjs/toolkit';
import createSagaMiddleware from '@redux-saga/core';

import imageReducer from 'store/ducks/images';
import sagas from 'store/sagas';

const sagaMiddleware = createSagaMiddleware();

const store = configureStore({
    devTools: true,
    reducer: {
        images: imageReducer
    },
    middleware: (getDefaultMiddleware) =>
        getDefaultMiddleware({ thunk: false }).concat(sagaMiddleware)
});

sagaMiddleware.run(sagas);

export default store;
