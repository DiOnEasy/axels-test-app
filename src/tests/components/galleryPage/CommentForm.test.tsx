import '@babel/preset-react';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';

import { CommentForm } from 'components';

import store from 'store';

describe('CommentForm', () => {
    it('CommentForm snapshot', () => {
        const { asFragment } = render(
            <Provider store={store}>
                <CommentForm />
            </Provider>
        );

        expect(asFragment()).toMatchSnapshot();
    });

    it('should render the form correctly', () => {
        render(
            <Provider store={store}>
                <CommentForm />
            </Provider>
        );

        expect(screen.getByPlaceholderText('Ваше имя')).toBeInTheDocument();
        expect(
            screen.getByPlaceholderText('Ваш комментарий')
        ).toBeInTheDocument();
        expect(screen.getByRole('button')).toBeInTheDocument();
    });
});
