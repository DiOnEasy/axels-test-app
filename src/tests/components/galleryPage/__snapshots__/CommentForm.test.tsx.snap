// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentForm CommentForm snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 MuiGrid-grid-sm-7 css-nuwxbb-MuiGrid-root"
  >
    <form>
      <div
        class="MuiFormControl-root MuiTextField-root sc-gsFSjX fraPGw css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r0:"
            name="name"
            placeholder="Ваше имя"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-ihdtdm"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root sc-gsFSjX fraPGw css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r1:"
            name="text"
            placeholder="Ваш комментарий"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-ihdtdm"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <button
        class="sc-iGgVNO cESqvO"
        type="submit"
      >
        Оставить комментарий
      </button>
    </form>
  </div>
</DocumentFragment>
`;
